crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1088 @@
+///     height: 10pt,
+///     fill: gradient.linear(red, blue, space: space.at(1))
+///   )[
+///     #space.at(0)
+///   ]
+/// }
+/// ```
+///
+/// ## Direction
+///
+/// Some gradients are sensitive to the direction of the gradient. For example, a
+/// linear gradient has an angle that determines the direction of the gradient. Instead
+/// of the traditional clockwise angle, Typst uses an anti-clockwise angle, with 0° being
+/// from left-to-right, 90° from top-to-bottom, 180° from right-to-left, and 270° from
+/// bottom-to-top.
+///
+/// ```example
+/// #set block(spacing: 0pt)
+/// #stack(
+///   dir: ltr,
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 0deg)),
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 90deg)),
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 180deg)),
+///   square(size: 50pt, fill: gradient.linear(red, blue, dir: 270deg)),
+/// )
+/// ```
+///
+/// ## Note on compatibility
+///
+/// Typst's gradients were designed to be widely compatible; however, in
+/// [PDF.js](https://mozilla.github.io/pdf.js/), the reader bundled with Firefox,
+/// gradients in `rotate` blocks may not be rendered correctly. This is a bug in
+/// PDF.js and not in Typst. Despite this, every type of gradient has been
+/// tested in every major PDF reader, and should work in most browsers as
+/// expected.
+///
+/// ## Presets
+///
+/// Typst also includes a number of preset color maps. In the following section, the
+/// list of available presets is given, along with a sample of each gradient and
+/// relevant comments. Most of these color maps are chosen to be color blind friendly.
+///
+/// ### Turbo
+///
+/// The [`turbo`]($gradient.turbo) gradient is a rainbow-like gradient that is
+/// perceptually uniform. Turbo is a gradient that takes an optional number of
+/// stops, which is set to 20 by default.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.turbo(10)))
+/// ```
+///
+/// ### Cividis
+///
+/// The [`cividis`]($gradient.cividis) gradient is a blue to gray to
+/// yellow gradient that is perceptually uniform. Cividis is a gradient
+/// that takes an optional number of stops, which is set to 20 by default.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.cividis(10)))
+/// ```
+///
+/// ### Rainbow
+///
+/// The [`rainbow`]($gradient.rainbow) gradient is a rainbow gradient that is
+/// **not** perceptually uniform, therefore it should only be used for decorative
+/// purposes and not for data visualization. Rainbow is a gradient that takes an
+/// optional number of stops, which is set to 20 by default. This color map is best
+/// used by setting the interpolation color space to [HSL]($color.hsl).
+///
+/// ❌ This color map is **not** suitable for data visualization.
```

Comment:
I would remove the ✅ ... messages everywhere and instead write a notice like this here:

```suggestion
/// The [`rainbow`]($gradient.rainbow) gradient cycles through the full
/// color spectrum. Rainbow is a gradient that takes an
/// optional number of stops, which is set to 20 by default. This color map is best
/// used by setting the interpolation color space to [HSL]($color.hsl).
///
/// **Attention:** The rainbow gradient is **not** suitable for data visualization. It is
/// **not** perceptually uniform, so the differences between values become unclear to your readers.
/// It should only be used for decorative purposes. 
```

Ignore the line breaks.